src/components/settings/ColorPickerSection.jsx

```diff
@@ -0,0 +1,218 @@
+import React, { useCallback, useEffect, useState } from "react"
+import PropTypes from "prop-types"
+import * as _ from "lodash"
+
+import ColorPicker from "./ColorPicker"
+import FormFieldColor from "./FormFieldColor"
+
+import elementStyles from "../../styles/isomer-cms/Elements.module.scss"
+
+const ColorPickerSection = ({ colors, saveChanges }) => {
+  const [currColors, setCurrColors] = useState(colors)
+  const [colorPicker, setColorPicker] = useState({
+    currentColor: "",
+    elementId: "",
+    oldColors: {},
+  })
+  const [colorPickerToggle, setColorPickerToggle] = useState(false)
+
+  useEffect(() => {
+    setCurrColors(colors)
+  }, [colors])
+  // event listener callback function that resets ColorPicker modal
+  // when escape key is pressed while modal is active
+  const escFunction = useCallback(
+    (event) => {
+      if (event.key === "Escape") {
+        setColorPickerToggle(false)
+      }
+    },
+    [setColorPickerToggle]
+  )
+
+  // event listener callback function that resets ColorPicker modal
+  // when mouse clicks on area outside of modal while modal is active
+  const clickFunction = useCallback(
+    (event) => {
+      let { target } = event
+      let { tagName } = target
+      // keep checking parent element until you hit a tagName of FORM
+      while (tagName !== "FORM") {
+        target = target.parentElement
+        tagName = target.tagName
+      }
+      // toggle only if descendant of colorModal
+      if (target.id !== "colorModal") {
+        setColorPickerToggle(false)
+      }
+    },
+    [setColorPickerToggle]
+  )
+
+  useEffect(() => {
+    if (colorPickerToggle) {
+      // setup escape key event listener to exit from ColorPicker modal
+      document.addEventListener("keydown", escFunction)
+      document.addEventListener("click", clickFunction)
+    } else {
+      // remove event listeners
+      document.removeEventListener("keydown", escFunction)
+      document.removeEventListener("click", clickFunction)
+      if (!_.isEmpty(colorPicker.oldColors)) {
+        setCurrColors(colorPicker.oldColors)
+      }
+    }
+    return () => {
+      document.removeEventListener("keydown", escFunction)
+      document.removeEventListener("click", clickFunction)
+    }
+  }, [colorPickerToggle, escFunction, clickFunction])
+
+  // toggles color picker modal
+  const activateColorPicker = (event) => {
```

Comment:
Is it possible to move these handlers (`activateColorPicker`, `onColorSelect`, `setRealTimeColor`, `changeHandler` etc) into `FormFieldColor` and make each component responsible for its own update?

Reply:
Good suggestion, updated in https://github.com/isomerpages/isomercms-frontend/pull/666/commits/1dd482ca0214a79ba30f7dee0fd993e9427efe91